server.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+)
+
+type User struct {
+	Host        string   `json:"host"`
+	User_agent  string   `json:"user_agent"`
+	Request_uri string   `json:"request_uri"`
+	Headers     *Headers `json:"headers"`
+}
+
+type Headers struct {
+	User_agent string `json:"User_agent"`
+	Accept     string `json:"Accept"`
+}
+
+func Handler(w http.ResponseWriter, r *http.Request) {
+	user := User{
+		r.Host,
+		r.Header.Get("User-agent"),
+		r.URL.Path,
+		&Headers{r.Header.Get("User-agent"),
+			r.Header.Get("Accept")},
+	}
+	userJson, err := json.Marshal(user)
+	if err != nil {
+		panic(err)
```

Comment:
Please do not use panic here. It means that in case of error in process of handling request - whole server will be stopped. 
Return an error and error code instead.